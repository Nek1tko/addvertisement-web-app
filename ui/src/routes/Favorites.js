import React, { useState, useEffect } from 'react';
import AdRecordsTable from "../components/AdRecordsTable";
import AuthService from '../services/auth.service';
import { Redirect } from "react-router-dom";
import axios from 'axios';
import authHeader from "../services/auth-header";

const API_URL = "http://localhost:8080/ad/favourites/";

const FavoritesImpl = props => {
    const [ads, setAds] = React.useState([]);
    const { _, userId, } = AuthService.getUser();

    useEffect(() => {
        axios
            .get(API_URL + userId, { headers: authHeader() })
            .then(res => {
                setAds(res.data);
            })
    }, [userId, setAds]);

    return <AdRecordsTable
        ads={ads}
        userId={userId}
        history={props.history}
    />;
}

const Favorites = props => {
    if (!AuthService.getUser()) {
        return <Redirect to="/login" />
    }

    return <FavoritesImpl history={props.history} />;
}

export default Favorites;
