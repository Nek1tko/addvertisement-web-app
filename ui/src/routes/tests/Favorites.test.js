import { mount, shallow } from 'enzyme';
import Favorites from '../Favorites';
import { BrowserRouter, Router } from 'react-router-dom';
import AuthService from '../../services/auth.service';
import { createBrowserHistory } from 'history'
import { act } from 'react-dom/test-utils';
import authHeader from "../../services/auth-header";
import axios from "axios";
import React from "react";

jest.mock("../../services/auth-header");
jest.mock("../../services/auth.service");
jest.mock('axios');

const mCategory = {
    id: 1,
    name: "mock category"
};

const mSubcategory = {
    id: 1,
    name: "mock subcategory",
    category: mCategory
};

const mMetro = {
    id: 1,
    name: "Улица Дыбенко",
};

const mUserId = 1;

const mAds = [
    {
        id: 1,
        description: "mock description",
        price: "123",
        name: "mock name",
        metro: mMetro,
        subCategory: mSubcategory,
        isActive: true,
        isFavourite: true,
        saler: { id: mUserId }
    },
    {
        id: 2,
        description: "mock description 2",
        price: "1234",
        name: "mock name 2",
        metro: mMetro,
        subCategory: mSubcategory,
        isActive: true,
        isFavourite: true,
        saler: { id: mUserId + 1 }
    },
];

describe('FavoritesTests', () => {
    let mHistory, wrapper

    beforeEach(() => {
        mHistory = {
            ...createBrowserHistory(),
            push: jest.fn()
        };
        AuthService.getUser.mockReturnValue({ user: "mockUser" });
        authHeader.mockReturnValue({ header: "mockHeader" });
        axios.get.mockResolvedValue({ data: mAds });
        wrapper = mount(
            <Router history={mHistory}>
                <Favorites />
            </Router>
        );
    })

    afterEach(() => {
        jest.clearAllMocks();
    })

    test('FavoritesRenderTest', () => {
        mount(
            <BrowserRouter>
                <Favorites />
            </BrowserRouter>
        );
    });

    test('FavoritesSnapshotTest', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('FavoritesNotAuthorizedTest', () => {
        AuthService.getUser.mockReturnValue(null);
        wrapper = mount(
            <Router history={mHistory}>
                <Favorites />
            </Router>);
    });
});
