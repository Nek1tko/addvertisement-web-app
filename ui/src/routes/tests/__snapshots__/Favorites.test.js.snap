// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesTests FavoritesSnapshotTest 1`] = `ReactWrapper {}`;
